src/client/Microsoft.Identity.Client/Internal/Requests/InteractiveRequestOrchestrator.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.UI;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+
+    /// <summary>
+    /// This class decides the workflow of an interactive request. The business rules are: 
+    /// 
+    /// 1. An IWebUi based request typically handles this, if UseBroker is set to false
+    ///     1.1. Create an authorization URI
+    ///     1.2. Get an auth code. 
+    ///     1.3. Look at the code - if it's a broker install uri, then go to 2. Otherwise continue.
+    ///     1.3. Exchange the auth code for tokens
+    ///     
+    /// 2. If UseBroker is set to true or AAD requires the use of a broker, then: 
+    ///     2.1. See if a broker app can handle the request. If not, redirect to an install page.
+    ///     2.2. Otherwise, invoke the broker and wait for it to respond with a token.
+    ///     
+    ///     Question: is MSAL supposed to cache the token from the broker? (on both Android and iOS?)
+    /// </summary>
+    internal class InteractiveRequestOrchestrator
```

Comment:
This class encapsualtes the complex logic above. Please review and let me know if it is wrong (also question on line 28).
Again notice that this class is much cleaner that the old InteractiveRequest - 1 ctor, 1 public method and all members are readonly